pkg/handler3/handler.go

```diff
@@ -222,27 +210,13 @@ func (o *OpenAPIV3Group) UpdateSpec(specBytes []byte) (err error) {
 	o.rwMutex.Lock()
 	defer o.rwMutex.Unlock()
 
-	specPb, err := ToV3ProtoBinary(specBytes)
-	if err != nil {
-		return err
-	}
-
-	specPbGz := toGzip(specPb)
-
-	specBytesETag := computeETag(specBytes)
-	specPbETag := computeETag(specPb)
-	specPbGzETag := computeETag(specPbGz)
-
-	lastModified := time.Now()
-
-	o.specBytes = specBytes
-	o.specPb = specPb
-	o.specPbGz = specPbGz
-
-	o.specBytesETag = specBytesETag
-	o.specPbETag = specPbETag
-	o.specPbGzETag = specPbGzETag
+	o.pbCache = o.pbCache.New(func() ([]byte, error) {
```

Comment:
I think this object `OpenAPIV3Group` should be the exact same object we have for v2. I don't see a good reason to have it twice. Can we also move that to the internal package and share the implementation?